dataRetriever/requestHandlers/headerRequestHandler.go

```diff
@@ -0,0 +1,71 @@
+package requestHandlers
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+)
+
+// ArgHeaderRequestHandler is the argument structure used to create a new a header request handler instance
+type ArgHeaderRequestHandler struct {
+	ArgBaseRequestHandler
+	NonceConverter typeConverters.Uint64ByteSliceConverter
+}
+
+type headerRequestHandler struct {
+	*baseRequestHandler
+	nonceConverter typeConverters.Uint64ByteSliceConverter
+}
+
+// NewHeaderRequestHandler returns a new instance of header request handler
+func NewHeaderRequestHandler(args ArgHeaderRequestHandler) (*headerRequestHandler, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &headerRequestHandler{
+		baseRequestHandler: createBaseRequestHandler(args.ArgBaseRequestHandler),
+		nonceConverter:     args.NonceConverter,
+	}, nil
+}
+
+func checkArgs(args ArgHeaderRequestHandler) error {
+	err := checkArgBase(args.ArgBaseRequestHandler)
+	if err != nil {
+		return err
+	}
+	if check.IfNilReflect(args.NonceConverter) {
```

Comment:
I think it will work with check.IfNil
The definition for typeConverters.Uint64ByteSliceConverter contains the function IsInterfaceNil

Reply:
correct, updated